core/src/main/scala/chisel3/internal/MonoConnect.scala

```diff
@@ -439,11 +439,29 @@ private[chisel3] object MonoConnect {
   }
 
   def probeDefine(
-    sourceInfo: SourceInfo,
-    sink:       Data,
-    source:     Data,
-    context:    BaseModule
+    sourceInfo:  SourceInfo,
+    sinkProbe:   Data,
+    sourceProbe: Data,
+    context:     BaseModule
   ): Unit = {
+
+    // Reify sink and source if they're views
+    val sink = if (isView(sinkProbe)) {
+      reifySingleData(sinkProbe).getOrElse(
+        throwException(
+          s"If a DataView contains a Probe, it must resolve to one Data. $sinkProbe does not meet this criteria."
+        )
+      )
+    } else sinkProbe
```

Comment:
If we merge https://github.com/chipsalliance/chisel/pull/4107 and rebase on it, you can change this to just:
```suggestion
    val sink = reifySingleData(sinkProbe).getOrElse(
      throwException(
        s"If a DataView contains a Probe, it must resolve to one Data. $sinkProbe does not meet this criteria."
      )
    )
```

Reply:
done and it works